src/test/java/hudson/plugins/promoted_builds/PromotedBuildRebuildParameterProviderTest.java

```diff
@@ -81,11 +81,7 @@ public void testRebuild() throws Exception {
         Assert.assertEquals(b1.getNumber(), pbpv1.getRun().getNumber());
 
         // rebuild it
-        JenkinsRule.WebClient wc = j.createWebClient();
-        HtmlPage page = wc.getPage(b2, "rebuild");
-        HtmlForm form = page.getFormByName("config");
-        j.submit(form);
-        j.waitUntilNoActivity();
+        FreeStyleBuild b3 = j.assertBuildStatusSuccess(p2.scheduleBuild2(0));
```

Comment:
This is defeating the purpose of the test, which exists solely to exercise rebuilder functionality.

Rather, the test should be adapted to work with POST requests. Try calling `JenkinsRule.WebClient#.withThrowExceptionOnFailingStatusCode(false)`.

Reply:
> Try calling `JenkinsRule.WebClient#withThrowExceptionOnFailingStatusCode(false)`.

Actually the trick is probably to follow https://github.com/jenkinsci/rebuild-plugin/pull/155 in calling `WebClient.createCrumbedUrl` with `HttpMethod.POST`.

Reply:
Thanks for the pointer to https://github.com/jenkinsci/rebuild-plugin/pull/155 .  I read it more carefully and saw that in at least one test, it was clicking the "Rebuild" button and then submitting the resulting form.  That seems to work in this case as well.

43e7046c4aa1bf1d4f36e324115ae8a509a6a081 is the proposed addition to the test that clicks the Rebuild button and then submits the resulting form.